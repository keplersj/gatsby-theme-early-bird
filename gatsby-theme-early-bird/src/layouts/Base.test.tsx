import * as React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery } from "gatsby";
import { HelmetProvider } from "react-helmet-async";
import Layout from "./Base";

beforeEach((): void => {
  (useStaticQuery as jest.Mock).mockImplementation((): object => ({
    site: {
      siteMetadata: {
        title: "",
        siteUrl: ""
      }
    }
  }));
});

describe("Base Layout", (): void => {
  it("renders correctly", (): void => {
    const context: { helmet?: any } = {};
    const tree = renderer
      .create(
        <HelmetProvider context={context}>
          <Layout>
            <span>Test</span>
          </Layout>
        </HelmetProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(context.helmet).toMatchSnapshot();
  });
});
