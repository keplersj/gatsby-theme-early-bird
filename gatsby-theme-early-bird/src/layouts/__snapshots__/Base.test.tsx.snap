// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Layout renders correctly 1`] = `
<head>
  Object {}
  Object {
    "lang": "en",
  }
  <meta
    content=""
    data-react-helmet={true}
    property="og:title"
  />
  <title
    data-react-helmet={true}
  >
    
  </title>
</head>

<main>
  <span>
    Test
  </span>
</main>
`;
